### PR TITLE
Replace ClusterBlast SVG generation from static python to dynamic javascript

### PR DESCRIPTION
**Pull request by SJShaw**
Due to browser CORS policies, viewing HTML results on local filesystems prevented these CB SVGs from displaying without running local HTTP servers (see #102).

These SVGs generated at analysis time have been replaced by SVGs drawn dynamically when viewing the HTML, when they're useful.

The numbers here are using NC_003888.3 as an example.

Note: the PR size is inflated by embedding the javascript library itself, this will be removed before merging

### Pros
- the SVGs always display
- the removal of the SVG files themselves (around 500 of them, or 40% of the total) also frees up around 18Mb (reducing the size of the final zip file by around 25%)
- more flexible and powerful interactions
- allows future interaction changes to be more easily added, along with more easily drawn (some examples below)
- the `pysvg-py3` dependency can be removed
- will make it much easier to update clusterblast SVGs for #697

### Cons
- the javascript results datafile increases by about 40%, from 9Mb to 12.9Mb, and this extra bandwidth is required even when never looking at the clusterblast output (after #700, this increase would be from 7.8Mb to 10.5Mb, which is only 1Mb more than current master)

That's not a particularly small con, but I think the price is worth it.

### New interactions/changes
- clicking a query gene now uses the same gene selection process as other visualisations, highlighting the selected gene in the region view
- double clicking reference clusters (not the labels, which are links) flips the directionality of the reference, for easier visual comparison when the predicted ordering isn't as desired
- percentage similarity text has been removed, due to so many users misinterpreting the number (this ought to be followed by changing the overview page and how it displays "Most similar cluster", but that's a separate PR)
- accession and location text have been removed from the SVG text, as user testing showed that very few people ever looked at them, however they're still shown in the selection dropdown and used by the outgoing links 

### Examples
Using NC_003888 region 10 (3,523,335 - 3,603,988):

Style, before:
image

Style, after:
image

Reference gene tooltips, before:
image

Reference gene tooltips, after:
image

### Additional things that would be nice
- some kind of visual linkage between query and reference genes in single-reference view
- colouring only the reference genes matching a selected query gene when selected (and vice versa)
- colouring only the matching genes between the query and a reference in single-reference view

There's probably more things that would be nice, if anyone wants to test it out and give feedback.

### CODE REVIEW

**Comment by kblin** [comment]
Looks good to me, I'll try it out locally in a bit.

**Review by kblin — DISMISSED** [review]
Looks good to me 👍

We need to rebase this anyway and get rid of the static JS file, maybe we can fix that comment while at it.

File: antismash/modules/clusterblast/data_structures.py [hunk 1]
```diff
@@ -134,6 +135,20 @@ def __init__(self) -> None:
...
+    def similarity(self) -> int:
+        """ The similarity of the pairing, as a percentage in the range 0 to 100 """
+        if self._similarity < 0:
+            raise ValueError("Similarity not yet set")
+        return self._similarity
+
+    @similarity.setter
+    def similarity(self, similarity: int) -> None:
+        if not 0 <= similarity <= 100:
+            raise ValueError(f"Similarity ({similarity}) out of bounds (0-1)")
```
[the remark this answers is not in the document]
Comment (SJShaw) [reply antismash/modules/clusterblast/data_structures.py]: Fixed in latest push.

**Comment by SJShaw** [comment]
Updated to what hopefully is the final version, with the removal of baked-in JS and instead fetching the new antismash-js version `0.14.0` with the usual mechanism.

**Review by kblin — APPROVED** [review]
Looks good to me 👍

**Pull request merged**